### PR TITLE
docs: update E2E pipeline process in developer docs

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**Reason for Change**:

Include a reference to the new "comment to trigger the pipeline" feature soon to be enabled on the E2E pipeline.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
Don't merge until we switch to the new pipeline

/hold

File: docs/community/developer-guide.md [hunk 1]
```diff
@@ -248,7 +248,7 @@ AKS Engine employs a Continuous Integration (CI) system that incorporates Azure
 The following steps constitute the AKS Engine CI pipeline:
 
 1. Contributor opens a Pull Request (PR) against the AKS Engine project
-1. The PR triggers an Azure DevOps job that
+1. A member of the AKS Engine team comments on the PR to trigger an Azure DevOps job that
```
Comment (tariq1890) [inline docs/community/developer-guide.md]: My most nit comment till date:
```suggestion
1. An AKS Engine team member comments on the PR to trigger an Azure DevOps job that
```

Comment (tariq1890) [reply docs/community/developer-guide.md]:
```suggestion
2. A member of the AKS Engine team comments on the PR to trigger an Azure DevOps job that
```

Comment (tariq1890) [inline docs/community/developer-guide.md]:
```suggestion
2. An AKS Engine team member comments on the PR to trigger an Azure DevOps job that
```

**Pull request merged**